api/pkg/apis/v1alpha1/managers/stage/stage-manager_test.go

```diff
@@ -262,6 +262,77 @@ func TestCampaignWithTwoCounterStageLoop(t *testing.T) {
 	// assert.Equal(t, "fake", status.Outputs["__site"])
 }
 
+func TestCampaignWithTriggersCounterStageLoop(t *testing.T) {
+	stateProvider := &memorystate.MemoryStateProvider{}
+	stateProvider.Init(memorystate.MemoryStateProviderConfig{})
+	manager := StageManager{
+		StateProvider: stateProvider,
+	}
+	manager.VendorContext = &contexts.VendorContext{
+		EvaluationContext: &coa_utils.EvaluationContext{},
+		SiteInfo: v1alpha2.SiteInfo{
+			SiteId: "fake",
+		},
+	}
+	manager.Context = &contexts.ManagerContext{
+		VencorContext: manager.VendorContext,
+		SiteInfo: v1alpha2.SiteInfo{
+			SiteId: "fake",
+		},
+	}
+	var status model.StageStatus
+	activation := &v1alpha2.ActivationData{
+		Campaign:             "test-campaign",
+		Activation:           "test-activation",
+		Stage:                "test",
+		ActivationGeneration: "1",
+		Inputs: map[string]interface{}{
+			"foo": 1,
+			"bar": 1,
+		},
+		Outputs:   nil,
+		Provider:  "providers.stage.counter",
+		Namespace: "fakens",
+	}
+	for {
+		status, activation = manager.HandleTriggerEvent(context.Background(), model.CampaignSpec{
+			SelfDriving: true,
+			FirstStage:  "test",
+			Stages: map[string]model.StageSpec{
+				"test": {
+					Provider:      "providers.stage.counter",
+					StageSelector: "${{$if($lt($output(test,foo), 5), test, test2)}}",
+					Inputs: map[string]interface{}{
+						"count": 1,
+						"foo":   "${{$trigger(foo, 0)}}",
+					},
+				},
+				"test2": {
+					Provider:      "providers.stage.counter",
+					StageSelector: "${{$if($lt($output(test2,foo), 10), test2, '')}}",
+					Inputs: map[string]interface{}{
+						"count.init": "${{$output(test, count)}}",
+						"count":      1,
+						"foo":        "${{$output(test, foo)}}",
+						"bar":        "${{$trigger(bar, -1)}}",
+						"bazz":       "${{$trigger(bazz, 0)}}",
+					},
+				},
+			},
+		}, *activation)
+
+		if activation == nil {
+			break
+		}
+	}
+	assert.Equal(t, v1alpha2.Done, status.Status)
+	assert.True(t, v1alpha2.Done.EqualsWithString(status.StatusMessage))
+	assert.Equal(t, int64(7), status.Outputs["count"])
+	assert.Equal(t, int64(10), status.Outputs["foo"])
+	assert.Equal(t, int64(2), status.Outputs["bar"])
+	assert.Equal(t, int64(0), status.Outputs["bazz"])
+}
```

Comment:
Could you add a few comments about why the output values should be these? And how  $trigger affects the result